modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -630,6 +674,57 @@ public CompletableFuture<Void> start() {
                 TimeUnit.MILLISECONDS
         );
 
+        scheduledTableLeaseUpdateExecutor.scheduleAtFixedRate(() -> {
+                    for (Map.Entry<ZonePartitionId, Set<ReplicationGroupId>> entry : zonePartIdToTablePartId.entrySet()) {
+                        ZonePartitionId repGrp = entry.getKey();
+
+                        ReplicaMeta meta = placementDriver.getLeaseMeta(repGrp);
+
+                        if (meta != null) {
+                            HashSet<ReplicationGroupId> diff = new HashSet<>(entry.getValue());
+                            diff.removeAll(meta.subgroups());
+
+                            if (meta.getLeaseholderId().equals(localNodeId) && !diff.isEmpty()) {
+                                LOG.info("New subgroups are found for existing lease [repGrp={}, subGroups={}]", repGrp, diff);
+
+                                try {
+                                    placementDriver.addSubgroups(repGrp, meta.getStartTime().longValue(), diff)
+                                            .thenCompose(unused -> {
+                                                ArrayList<CompletableFuture<?>> requestToReplicas = new ArrayList<>();
+
+                                                for (ReplicationGroupId partId : diff) {
+                                                    EmptyPrimaryReplicaRequest req = REPLICA_MESSAGES_FACTORY.emptyPrimaryReplicaRequest()
+                                                            .enlistmentConsistencyToken(meta.getStartTime().longValue())
+                                                            .groupId(partId)
+                                                            .build();
+
+                                                    CompletableFuture<Replica> replicaFut = replicas.get(repGrp);
+
+                                                    if (replicaFut != null) {
+                                                        requestToReplicas.add(replicaFut.thenCompose(
+                                                                replica -> replica.processRequest(req, localNodeId)));
+                                                    }
+                                                }
+
+                                                return allOf(requestToReplicas.toArray(CompletableFuture[]::new));
+                                            }).join();
```

Comment:
Is join here required? why not get with timeout?

Reply:
I add a specific timeout here.